test/test_attn.py

```diff
@@ -0,0 +1,68 @@
+import torch
+from torch.autograd import gradcheck, Function
+from typing import Tuple
+
+
+# This file contains the custom autograd function of 'attn' for the purpose
+# of verifying that the manually derived gradients are correct
+
+class AttentionFunction(Function):
+
+    @staticmethod
+    def forward(ctx, q: torch.Tensor, k: torch.Tensor, v: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        k_t = k.t()
+        mm_output = torch.mm(q, k_t)
+        tanh_output = torch.tanh(mm_output)
+        output = torch.mm(tanh_output, v)
+
+        ctx.save_for_backward(q, k, v, mm_output, tanh_output)
```

Comment:
As mentioned offline, using mm_output here means that `gradgradcheck` will fail. You want to fix that I think.

Reply:
Fixed.